### PR TITLE
fix message placement and rounded corners on API 21/22

### PR DESCRIPTION
**Pull request by rymorale**
## Description
- fix message placement and rounded corners on API 21/22

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by PravinPK — APPROVED** [review]

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ class AEPMessage implements FullscreenMessage {
     // private vars
     private WebView webView;
     private CardView webViewFrame;
-    private ViewGroup.LayoutParams params;
+    private FrameLayout.LayoutParams params;
```
Comment (rymorale) [inline code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java:71]: CardView extends FrameLayout directly. Using FrameLayout LayoutParams resolves the incorrect placement of messages on Android API 22 and below.

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java [hunk 2]
```diff
@@ -385,9 +385,9 @@ void cleanup(final boolean dismissedWithBackTouch) {
 
         if (dismissedWithBackTouch) {
             listener.onBackPressed(this);
-        } else {
-            listener.onDismiss(this);
         }
+        listener.onDismiss(this);
```
Comment (rymorale) [inline code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java:389]: do dismiss tracking as well if back press is detected

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/MessageWebViewUtil.java [hunk 3]
```diff
@@ -152,6 +157,21 @@ void show(final AEPMessage message) {
...
+                    TypedValue.applyDimension(
+                            TypedValue.COMPLEX_UNIT_DIP,
+                            message.getMessageSettings().getCornerRadius(),
+                            context.getResources().getDisplayMetrics());
+            roundedDrawable.setCornerRadius(calculatedRadius);
+            webViewFrame.setBackground(roundedDrawable);
+
+            // if API < 22 then set webview alpha to 99% to fix rounded corners on messages
+            if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.LOLLIPOP_MR1) {
+                webView.setAlpha(WORKAROUND_ALPHA_VALUE);
```
[the remark this answers is not in the document]
Comment (prudrabhat) [reply code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/MessageWebViewUtil.java:172]: Is this a known issue/hack that is applied for lower android versions generally? Curious about the rootcause here.

[the remark this answers is not in the document]
Comment (rymorale) [reply code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/MessageWebViewUtil.java:172]: this hack was found by observing that the parent cardview was sometimes visible under the displayed webview when the display / dismiss animation was occurring. this led me to start looking into modifying the transparency of the webview and found that 99 percent transparency caused the webview to be framed by the parent cardview.

[the remark this answers is not in the document]
Comment (prudrabhat) [reply code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/MessageWebViewUtil.java:172]: Hmm, I am a bit unclear on whether this is a bug in our code (perhaps incorrectly using a documented api for older versions) or in the android subsystem itself. Let us discuss offline before we merge these changes.

[the remark this answers is not in the document]
Comment (prudrabhat) [reply code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/MessageWebViewUtil.java:172]: One last try. Based [on this answer](https://stackoverflow.com/a/11927425) it looks like the webview has a default background color of white, drawn in front of any drawables. Can you try `webview.setBackgroundColor(0);` to check if it solves the issue. While prototyping the new UIServices ui this trick worked for me.

[the remark this answers is not in the document]
Comment (rymorale) [reply code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/MessageWebViewUtil.java:172]: The webview background color is already set to 0 / transparent when it is being created at https://github.com/adobe/aepsdk-core-android/blob/dev/code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java#L486. Setting the background color to 0 within the `MessageWebViewUtil` class does not resolve the rounded corner issue on API 21/22.

File: code/core/src/main/java/com/adobe/marketing/mobile/internal/CoreConstants.kt [hunk 4]
```diff
@@ -13,7 +13,7 @@ package com.adobe.marketing.mobile.internal
 
 internal object CoreConstants {
     const val LOG_TAG = "MobileCore"
-    const val VERSION = "2.2.2"
+    const val VERSION = "2.2.3"
```
Comment (prudrabhat) [inline code/core/src/main/java/com/adobe/marketing/mobile/internal/CoreConstants.kt:16]: I am assuming these changes are due to the previous release was directly pushed to main.

**Review by prudrabhat — APPROVED** [review]

File: code/core/src/main/java/com/adobe/marketing/mobile/internal/CoreConstants.kt (hunk 4 above)
Comment (rymorale) [reply code/core/src/main/java/com/adobe/marketing/mobile/internal/CoreConstants.kt:16]: yes, this pr has the changes from the lightroom npe fix branch

**Pull request merged**